PBA-5502 Issues with Closed Captions

orobles-ooyala opened the pull request:
- Wrap closed captions to the next line.
- Do not scale closed captions text size when rotating the device, instead
  use a fixed 16 size.
- Place the closed captions at the same level as the watermark, and extend
  the closed captions to the full width if there is no watermark.
- Allow the closed caption menu to resize according to the player width,
  use a ScrollBack as a fallback if the height is not enough to show all
  the buttons.

orobles-ooyala committed:
PBA-5502 Issues with Closed Captions

- Wrap closed captions to the next line.
- Do not scale closed captions text size when rotating the device, instead
  use a fixed 16 size.
- Place the closed captions at the same level as the watermark, and extend
  the closed captions to the full width if there is no watermark.
- Allow the closed caption menu to resize according to the player width,
  use a ScrollBack as a fallback if the height is not enough to show all
  the buttons.

brol1dev commented on sdk/react/panels/style/languageSelectionPanelStyles.json:
```diff
@@ -55,6 +55,10 @@
     "alignSelf": "stretch",
     "backgroundColor": "#585858"
   },
+  "selectionPreviewtextContainer": {
```
Follow camel case for all the words, it should be `selectionPreviewTextContainer`. Can it be simpler? e.g. `previewTextContainer`

orobles-ooyala replied:
That was a typo :( , I will change it to previewTextContainer

brol1dev commented on sdk/react/panels/languageSelectionPanel.js:
```diff
@@ -117,35 +154,29 @@ var LanguageSelectionPanel = React.createClass({
-          itemHeight={88}>
-        </ResponsiveList>
-        {previewText}
+        <ScrollView>
```
I think the ScrollView shouldn't wrap the header so it is always visible. Although I can see why you did it, is it because on smaller screens there is not enough space?

What do you think about having the `renderHeader` outside of the `ScrollView`?

orobles-ooyala replied:
It make sense, I will do some test to check if everything is ok

brol1dev reviewed (commented):
Looks great, just added some inline comments.

Question, what are the styles you are using to wrap the CCs to the next line if they are too long?

brol1dev commented:
By the way, I just noticed there is a conflict with the `languageSelectionPreview.js` file, can you merge the latest changes in dev into this branch and fix the issues? Thanks.

orobles-ooyala commented:
**Question, what are the styles you are using to wrap the CCs to the next line if they are too long?**

That is the default behavior unless a "flexDirection":"row" is given

orobles-ooyala committed:
PBA-5502 Minor fix to closed captions

- Do not show the title when the localized text length is longer
  than 10 and will not fit the screen.
- Move the header out of the ScrollView

brol1dev commented:
LGTM
